src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -939,7 +946,14 @@ public void getModifiedDocIds(DocIdPusher pusher)
         hasTimestamp = hasColumn(rsmd, GsaSpecialColumns.GSA_TIMESTAMP);
         log.log(Level.FINEST, "Has GSA_TIMESTAMP column: {0}", hasTimestamp);
         while (rs.next()) {
-          DocId id = new DocId(uniqueKey.makeUniqueId(rs));
+          DocId id;
+          try {
+            id = new DocId(uniqueKey.makeUniqueId(rs));
+          } catch (URISyntaxException e) {
+            log.log(Level.WARNING, "Invalid DocId URL, skipping row: "
+                + e.getMessage());
+            continue;
+          }
```

Comment:
Yes, it could, at the expense of another line of code and another local variable clouding up a very small block.  new DocId() throws no exceptions, so this was as small a reasonable try/catch block I was willing to go.